service/src/main/java/org/exoplatform/addons/lecko/SimpleDataBuilder.java

```diff
@@ -317,7 +317,12 @@ public boolean build() {
   public void deleteDumpFile() {
     String extractOutputPath = leckoTempDirectory + "/" + leckoOutputName;
     File file = new File(extractOutputPath);
-    file.delete();
+    try {
+        file.delete();
+    }
+    catch (Exception ex){
+        LOG.info("Unable to delete the dump file: ", ex);
+    }
```

Comment:
Do we really need this try/catch ? We saw that the delete() method does not throw an exception even when if fails. Instead it returns false. So we should check if boolean instead IMO. WDYT ?

Reply:
When I tried to get the value of file.delete(), it was always **false** even the dump file was deleted. That's why I didn't keep the return value of file.delete(). 

![Screenshot from 2019-03-12 17-14-28](https://user-images.githubusercontent.com/40362165/54216634-4f00cf80-44ea-11e9-9db2-3e39ef2aeb01.png)

Reply:
If the method returns false, it means that the file does not exist or that the file/folder could ot be deleted, that's weird.
We should probably switch to new Files API : https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#delete(java.nio.file.Path)
It is better and gives more info when there are errors.
WDYT ?

Reply:
okey, but I think that we will use _Files.deleteIfExist(path)_ instead of _Files.delete(path)_ because it returns a boolean while Files.delete is a void method.

```suggestion
    String extractOutputPath = leckoTempDirectory + "/" + leckoOutputName;
    Path path = Paths.get(extractOutputPath);
    try{
       if(Files.deleteIfExists(path))
       {
           LOG.info("Dump file deleted.");
       }
       else {
           LOG.warn("Dump file is not found. Status: "+Files.deleteIfExists(path));
       }
    }
    catch (Exception ex){
       LOG.error("Trying to delete the dump file. Deleting Status: "+Files.exists(path),ex);
    }
```

Reply:
Both methods are ok for me. Files.delete throw an NoSuchFileException if the file does not exist whereas Files.deleteIfExist returns false.

Reply:
Yeah, I already tested with both methods.